### PR TITLE
Update libs. Worker init wait.

### PR DESCRIPTION
**Pull request by rpiaggio**
- Upgrades to Scala 3 enums, latest lucuma-core and lucuma-ui.
- Mechanism to wait for worker server initialization.

This PR merges changes from #2133

### CODE REVIEW

File: workers/src/main/scala/workers/WorkerServer.scala [hunk 1]
```diff
@@ -109,4 +110,12 @@ trait WorkerServer[F[_]: Async, T: Pickler](using Monoid[F[Unit]]) {
...
     ).handleErrorWith(e => Logger[F].error(e)("Error initializing worker"))
-}
+
+  protected def runInternal(dispatcher: Dispatcher[F])(using Logger[F]): F[Unit] =
+    for {
+      self         <- F.delay(dom.DedicatedWorkerGlobalScope.self)
+      handlerFn    <- handler
+      cancelTokens <- Ref[F].of(Map.empty[WorkerProcessId, F[Unit]])
+      _            <- mount(self, handlerFn, cancelTokens)(dispatcher)
+      _            <- postAsTransferable[F, FromServer](self, FromServer.Ready)
```
Comment (cquiroz) [inline workers/src/main/scala/workers/WorkerServer.scala:120]: Will clients wait for this message?

Comment (rpiaggio) [reply workers/src/main/scala/workers/WorkerServer.scala:120]: Yup, it's done here https://github.com/gemini-hlsw/explore/blob/move-to-ui/common/src/main/scala/workers/WorkerClient.scala#L27

File: common/src/main/scala/workers/WorkerClient.scala [hunk 2]
```diff
@@ -76,4 +96,12 @@ class WorkerClient[F[_]: Concurrent: UUIDGen: Logger, R: Pickler](worker: WebWor
...
   ): F[Unit] =
     request(requestMessage).use(_.compile.drain)
-}
+
+object WorkerClient:
+  def fromWorker[F[_]: Concurrent: UUIDGen: Logger, R: Pickler](
+    worker: WebWorkerF[F]
+  ): Resource[F, WorkerClient[F, R]] =
+    for {
+      latch  <- Resource.eval(Deferred[F, Unit])
```
Comment (cquiroz) [inline common/src/main/scala/workers/WorkerClient.scala:105]: What are we using the latch for?

File: explore/src/main/scala/explore/constraints/ConstraintsPanel.scala [hunk 3]
```diff
@@ -42,30 +42,23 @@ import react.semanticui.collections.form.Form
...
-    case object HourAngle extends ElevationRangeType("Hour Angle")
-
-    implicit val enumeratedElevationRangeType: Enumerated[ElevationRangeType] =
-      Enumerated.of(AirMass, HourAngle)
-
-    implicit val displayElevationRangeType: Display[ElevationRangeType] =
-      Display.byShortName(_.label)
-  }
+  private object ElevationRangeType:
+    given Enumerated[ElevationRangeType] = Enumerated.from(AirMass, HourAngle).withTag(_.label)
```
Comment (cquiroz) [inline explore/src/main/scala/explore/constraints/ConstraintsPanel.scala:60]: Maybe this one can be derived too?

File: explore/src/main/scala/explore/targeteditor/RVInput.scala [hunk 4]
```diff
@@ -34,26 +35,14 @@ final case class RVInput(
...
+    case Z  extends RVView("z".refined)
+    case CZ extends RVView("cz".refined)
 
-    implicit val rvViewEnumeration: Enumerated[RVView] =
-      Enumerated.of(RV, Z, CZ)
-
-    implicit val rvDisplay: Display[RVView] = Display.by(_.tag.value, _.tag.value)
-  }
+  private object RVView:
+    given Enumerated[RVView] = Enumerated.from(RVView.RV, RVView.Z, RVView.CZ).withTag(_.tag)
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/RVInput.scala:44]: Can it be derived?

**Review by cquiroz — APPROVED** [review]
LGTM thanks a lot

File: explore/src/main/scala/explore/constraints/ConstraintsPanel.scala (hunk 3 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/constraints/ConstraintsPanel.scala:60]: Yes, there may be others too. The goal here wasn't to migrate everything yet, just to get things to compile :D

File: common/src/main/scala/workers/WorkerClient.scala (hunk 2 above)
Comment (rpiaggio) [reply common/src/main/scala/workers/WorkerClient.scala:105]: The latch is released when the `Ready` message is received from the server. This allows us to wait asynchronously for the server to report that it's up. In line 50 we wait for the latch before sending a request.

I tried the `WorkerClient` initialization to wait synchronously for the server but explore took noticeably longer to start up.

File: explore/src/main/scala/explore/constraints/ConstraintsPanel.scala (hunk 3 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/constraints/ConstraintsPanel.scala:60]: Oh wait, the derivation macro doesn't work in nested `enum`s. I couldn't figure it out, but there's probably a way.

In these cases we can either move the `enum` out of wherever it's nested, or manually declare the `Enumerated`.

Comment (rpiaggio) [reply explore/src/main/scala/explore/constraints/ConstraintsPanel.scala:60]: Let's do a separate PR changing the remaining `Enumerated` declarations to `enum` + `derives`.

**Pull request merged**